src/models/BaseModel.py

```diff
@@ -1,118 +1,190 @@
-            self.cleaned_region_feature = train_normalized_location_data[['region']]
-
-    @staticmethod
-    def load_location_data(train_normalized_location_file_name):
-        BaseModel.normalized_location_data = pd.read_csv(train_normalized_location_file_name)
+    TRAIN_NORMALIZED_LOCATION_FILE_NAME = '../data/train_normalised_location.csv'
+    TRAIN_DATA_CSV_FILE_NAME = '../data/Train_rev1.csv'
+    CLEANED_FILE_NAME = '../data/Preprocessed_Data.csv'
+
+    def __init__(self, train_size=0.75):
+        print('Initialized BaseModel')
+        if os.path.exists(BaseModel.CLEANED_FILE_NAME):
+            print('Pre-processed data exists, reading from the file')
+            self.cleaned_encoded_data = self.load_cleaned_data()
+            print('Data read complete')
+        else:
+            print('Pre-processed data doesn\'t exist, preprocessing data first')
+            print('This operation will take a while')
+            self.load_all_data()
+            print('Raw data read complete')
+            # Get features to clean and reshape
+            self.description_feature = BaseModel.data[['FullDescription']]
+            self.contract_type_feature = BaseModel.data[['ContractType']]
+            self.contract_time_feature = BaseModel.data[['ContractTime']]
+            self.category_feature = BaseModel.data[['Category']]
+            self.company_feature = BaseModel.data[['Company']]
+            self.source_name_feature = BaseModel.data[['SourceName']]
+            self.location_raw_feature = BaseModel.data[['LocationRaw']]
+            self.title_feature = BaseModel.data[['Title']]
+            self.salary_feature = BaseModel.data[['SalaryNormalized']]
+            # Read cleaned locations from separate file, cleaned with Google location
+            self.cleaned_town_feature = BaseModel.normalized_location_data[['town']]
+            self.cleaned_region_feature = BaseModel.normalized_location_data[['region']]
+
+            (self.processed_data, self.feature_list) = self.preprocess_data()
+            self.export_data(self.processed_data, 'Preprocessed_Data')
+            self.export_data(self.feature_list, 'Feature_List')
+            self.cleaned_encoded_data = self.load_cleaned_data()
+
+        print('Splitting train and test')
+        self.train_data, self.test_data = train_test_split(self.cleaned_encoded_data,
+                                                           train_size=train_size, random_state=9)
+        # Because 9 is good
+        # Random state is there so that train and test is always the same for everyone
+        self.x_train = self.train_data[:, 0:self.train_data.shape[1] - 1]
+        self.y_train = self.train_data[:, self.train_data.shape[1] - 1]
+
+        self.x_test = self.test_data[:, 0:self.test_data.shape[1] - 1]
+        self.y_test = self.test_data[:, self.test_data.shape[1] - 1]
+        print('Train test split complete \n')
+
+    def preprocess_data(self):
```

Comment:
This is where all the stuff in old main went into, all the cleaning we have done so far is here.